### ISSUE TITLE
Issue with Tag Deserialization in Cooked game

### ISSUE THREAD

**Issue by Tryibion** [FlaxEngine/FlaxEngine#1534]
**Issue description:**
Tags do not seem to deserialize correctly in a cooked game. It seems to work fine in the editor. This is still happening after the Tag null fixes submitted.

```
[ 00:00:01.523 ]: [Warning] Inner exception. Could not cast or convert from System.String to FlaxEngine.Tag.
Stack strace:
   at Newtonsoft.Json.Utilities.ConvertUtils.EnsureTypeAssignable(Object value, Type initialType, Type targetType) in F:\FlaxEngineDotnet\Cache\Intermediate\Deps\NewtonsoftJson\Src\Newtonsoft.Json\Utilities\ConvertUtils.cs:line 641
   at Newtonsoft.Json.Utilities.ConvertUtils.ConvertOrCast(Object initialValue, CultureInfo culture, Type targetType) in F:\FlaxEngineDotnet\Cache\Intermediate\Deps\NewtonsoftJson\Src\Newtonsoft.Json\Utilities\ConvertUtils.cs:line 612
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.EnsureType(JsonReader reader, Object value, CultureInfo culture, JsonContract contract, Type targetType) in F:\FlaxEngineDotnet\Cache\Intermediate\Deps\NewtonsoftJson\Src\Newtonsoft.Json\Serialization\JsonSerializerInternalReader.cs:line 1013

[ 00:00:01.523 ]: [Warning] Exception has been thrown during ManagedSerialization::Deserialize. Error converting value "Player" to type 'FlaxEngine.Tag'. Path 'TestTags[0]', line 1, position 170.
Stack strace:
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.EnsureType(JsonReader reader, Object value, CultureInfo culture, JsonContract contract, Type targetType) in F:\FlaxEngineDotnet\Cache\Intermediate\Deps\NewtonsoftJson\Src\Newtonsoft.Json\Serialization\JsonSerializerInternalReader.cs:line 1017
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateValueInternal(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue) in F:\FlaxEngineDotnet\Cache\Intermediate\Deps\NewtonsoftJson\Src\Newtonsoft.Json\Serialization\JsonSerializerInternalReader.cs:line 335
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.PopulateList(IList list, JsonReader reader, JsonArrayContract contract, JsonProperty containerProperty, String id) in F:\FlaxEngineDotnet\Cache\Intermediate\Deps\NewtonsoftJson\Src\Newtonsoft.Json\Serialization\JsonSerializerInternalReader.cs:line 1719
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateList(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, Object existingValue, String id) in F:\FlaxEngineDotnet\Cache\Intermediate\Deps\NewtonsoftJson\Src\Newtonsoft.Json\Serialization\JsonSerializerInternalReader.cs:line 898
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateValueInternal(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue) in F:\FlaxEngineDotnet\Cache\Intermediate\Deps\NewtonsoftJson\Src\Newtonsoft.Json\Serialization\JsonSerializerInternalReader.cs:line 313
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.SetPropertyValue(JsonProperty property, JsonConverter propertyConverter, JsonContainerContract containerContract, JsonProperty containerProperty, JsonReader reader, Object target) in F:\FlaxEngineDotnet\Cache\Intermediate\Deps\NewtonsoftJson\Src\Newtonsoft.Json\Serialization\JsonSerializerInternalReader.cs:line 1064
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.PopulateObject(Object newObject, JsonReader reader, JsonObjectContract contract, JsonProperty member, String id) in F:\FlaxEngineDotnet\Cache\Intermediate\Deps\NewtonsoftJson\Src\Newtonsoft.Json\Serialization\JsonSerializerInternalReader.cs:line 2479
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.Populate(JsonReader reader, Object target) in F:\FlaxEngineDotnet\Cache\Intermediate\Deps\NewtonsoftJson\Src\Newtonsoft.Json\Serialization\JsonSerializerInternalReader.cs:line 116
   at Newtonsoft.Json.JsonSerializer.PopulateInternal(JsonReader reader, Object target) in F:\FlaxEngineDotnet\Cache\Intermediate\Deps\NewtonsoftJson\Src\Newtonsoft.Json\JsonSerializer.cs:line 824
   at Newtonsoft.Json.JsonSerializer.Populate(JsonReader reader, Object target) in F:\FlaxEngineDotnet\Cache\Intermediate\Deps\NewtonsoftJson\Src\Newtonsoft.Json\JsonSerializer.cs:line 797
   at FlaxEngine.Json.JsonSerializer.Deserialize(Object input, Byte* jsonBuffer, Int32 jsonLength) in Z:\Flax\Source\FlaxEngine\Source\Engine\Serialization\JsonSerializer.cs:line 328
   at FlaxEngine.Interop.NativeInterop.Invoker.InvokerStaticNoRet3`3.MarshalAndInvoke(Object delegateContext, ManagedHandle instancePtr, IntPtr paramPtr) in Z:\Flax\Source\FlaxEngine\Source\Engine\Engine\NativeInterop.Invoker.cs:line 722
   at FlaxEngine.Interop.NativeInterop.InvokeMethod(ManagedHandle instanceHandle, ManagedHandle methodHandle, IntPtr paramPtr, IntPtr exceptionPtr) in Z:\Flax\Source\FlaxEngine\Source\Engine\Engine\NativeInterop.Unmanaged.cs:line 704
[ 00:00:01.523 ]: [Error] Exception has been thrown during ManagedSerialization::Deserialize.
Error converting value "Player" to type 'FlaxEngine.Tag'. Path 'TestTags[0]', line 1, position 170.
[ 00:00:01.619 ]: [Warning] Invalid or missing asset (01f8137b44b8e3ca11c9eaa58d9309a8, FlaxEngine.Animation).
[ 00:00:01.629 ]: [Warning] Exception has been thrown. Object reference not set to an instance of an object.
Stack strace:
   at Game.TestUI.OnStart() in C:\Users\mrcha\Documents\Flax Projects\TestingProject\Source\Game\Scripts\TestUI.cs:line 53
   at FlaxEngine.Interop.NativeInterop.Invoker.InvokerNoRet0`1.InvokeThunk(Object delegateContext, ManagedHandle instancePtr, IntPtr* paramPtrs) in Z:\Flax\Source\FlaxEngine\Source\Engine\Engine\NativeInterop.Invoker.cs:line 301
   at FlaxEngine.Interop.NativeInterop.ThunkContext.InvokeThunk(ManagedHandle instanceHandle, IntPtr param1, IntPtr param2, IntPtr param3, IntPtr param4, IntPtr param5, IntPtr param6, IntPtr param7) in Z:\Flax\Source\FlaxEngine\Source\Engine\Engine\NativeInterop.cs:line 1335
[ 00:00:01.629 ]: [Warning] Exception has been thrown.
Object reference not set to an instance of an object.

```

**Steps to reproduce:**
Make a Tag or Tag[] member variable. Cook the game. See error. This does stop anything else on the script/actor from deserializing causing the code not to run correctly.

**Flax version:**
1.6.6345

**Comment by walterman105** [comment]
This also happens to me, also on Flax version 1.6.6345.

**Comment by mafiesto4** [comment]
Fixed in b492c9421bc7b73245f394a48335d2dcee553034.
